### ISSUE TITLE
image mask is not removed when I send an image to it

### ISSUE THREAD

**Issue by lujiazho** [gradio-app/gradio#3444]
### Describe the bug

```
sendtobtn = gr.Button("SendOutToInput")        
sendtobtn.click(sendto, inputs=[out_image], outputs=[input_image])
```

I use the above code to send the generated image from the out Image component to the input Image component. The input and output images are defined:

```
input_image = gr.Image(source='upload', type='numpy', interactive=True, tool='sketch', elem_id="image_upload", brush_radius=200).style(height=400)
out_image = gr.Image(label="result", interactive=False).style(height=400)
```

When I send the output image to input image, the image in input component is changed and mask seems gone as expected. But when I start to run with the previous output image as new input, the new result is affected by the previously drawn mask.

And I checked the mask, the previously drawn mask is still there. How can I solve this?

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Reproduction

```
def sendto(x):
    return x

input_image = gr.Image(source='upload', type='numpy', interactive=True, tool='sketch', elem_id="image_upload", brush_radius=200).style(height=400)
out_image = gr.Image(label="result", interactive=False).style(height=400)

sendtobtn = gr.Button("SendOutToInput")        
sendtobtn.click(sendto, inputs=[out_image], outputs=[input_image])
```

### Logs

```shell
No error logs
```

### System Info

```shell
gradio 3.20.1
Windows 10
Chrome
```

### Severity

annoying

**Comment by everythingfades** [comment]
same issue with similar codes, this would also happen with matplotlib images and annotations.

**Comment by everythingfades** [comment]
solved the issue for matplotlib images by redefining fig,ax = plt.subplot().

**Comment by space-nuko** [comment]
Working repro:

```python
#!/usr/bin/env python

import gradio as gr
from PIL import Image

def sendto(x):
    print(x)
    im = x["image"].copy()
    im.paste(x["mask"], (0, 0), x["mask"])
    return im

with gr.Blocks() as demo:
    with gr.Column():
        input_image = gr.Image(source='upload', type='pil', interactive=True, tool='sketch', elem_id="image_upload", brush_radius=200).style(height=400)
        ext_image = gr.Image(source="upload", type='pil', label="out", interactive=True).style(height=400)

    result_image = gr.Image(label="result", interactive=False).style(height=400)

    renderbtn = gr.Button("Render")
    renderbtn.click(sendto, inputs=[input_image], outputs=[result_image])

    sendtobtn = gr.Button("SendOutputToInput")
    sendtobtn.click(lambda x: x, inputs=[ext_image], outputs=[input_image])

demo.queue().launch(server_port=9876)
```

Steps to reproduce:
1. Set two different images in the image inputs
2. Draw a mask on the first image
3. Click "Render" and see the output
4. Click "SendOutputToInput", mask should be cleared
5. Draw a mask on the first image again
6. Click "Render" and see the output
7. Now click "SendOutputToInput" again, mask should not be cleared
8. Click "Render" and this error appears

```
Traceback (most recent call last):
  File "C:\Users\a\build\gradio-test\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\Users\a\build\gradio-test\venv\lib\site-packages\gradio\blocks.py", line 1073, in process_api
    inputs = self.preprocess_data(fn_index, inputs, state)
  File "C:\Users\a\build\gradio-test\venv\lib\site-packages\gradio\blocks.py", line 962, in preprocess_data
    processed_input.append(block.preprocess(inputs[i]))
  File "C:\Users\a\build\gradio-test\venv\lib\site-packages\gradio\components.py", line 1590, in preprocess
    assert isinstance(x, dict)
AssertionError
```

Only issue is, with this repro I can't get the mask to disappear but still take effect, that was the issue I was coming across in webui. In that case the image the backend receives isn't a dict with `image` and `mask` keys but a plain `PIL.Image` instead

**Comment by abidlabs** [comment]
Removing from 3.x milestone as we will likely rework the Image component as part of 4.0

**Comment by pngwn** [comment]
Fixed by #6169.
